portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -72,4 +77,42 @@ private Key getUserKey(String userId) {
     Entity userEntity = results.asSingleEntity();
     return userEntity.getKey();
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
Either leave this out or mark it as a "TODO" (and it's fine if you don't actually end up getting to it, as long as it is a topic you intend to come back to a TODO is appropriate).  If you aren't sure whether you want to do this you can put in a TODO to assess
e.g. TODO: Investigate options for image validation.